src/Illuminate/Cache/Repository.php

```diff
@@ -175,9 +177,11 @@ protected function handleManyResult($keys, $key, $value)
     /**
      * Retrieve an item from the cache and delete it.
      *
+     * @template TCached of mixed
+     *
      * @param  string  $key
-     * @param  mixed  $default
-     * @return mixed
+     * @param  TCached  $default
```

Comment:
This should consideration that the `$default` value may be a closure that returns `TCacheValue`. And this comment can applied to other arguments annotations in this pull request.

Reply:
Wasn't aware of this possibility in Laravel, hopefully this will help more discover it as well :)